### DISCUSSION TITLE
Home page routing

### DISCUSSION THREAD

**Discussion by scottiemc7** [OrchardCMS/OrchardCore#9091] [Q&A] [answered]
Hi. I am trying to get a different home page to load depending on the host header sent in the request. Basically it would be a dynamic verison of the HomeRoute module. I created a class inheriting from `DynamicRouteValueTransformer` and registered it in Startup.

    public override ValueTask<RouteValueDictionary> TransformAsync(HttpContext httpContext, RouteValueDictionary values)
    {
        if (httpContext.Request.Headers.TryGetValue("Host", out var host) && host.First().StartsWith("testme.local"))
        {
            return ValueTask.FromResult(new RouteValueDictionary()
            {
                { "Action", "Display" },
                { "Controller", "Item" },
                { "Area", "OrchardCore.Contents" },
                { "ContentItemId", "47hxm0xbps87r3edx97bssfh50" }
            });
        }
        
        return ValueTask.FromResult(new RouteValueDictionary()
        {
            { "Page", "/App/Index" },
            { "Area", "MyAppArea" }
        });
    }

When I visit testme.local it brings up the content item which is great. However, if I try localhost it returns a 404. I can't seem to route to any razor page in my main app. I can visit /App/Index directly and verify that the route values are correct. What am I doing wrong?

Edit: Or maybe this is just a really dumb idea? 🤷  Am I asking for trouble?

**Comment by ns8482e** [comment, marked as answer]
what is your `Configure` method in Startup look like? I feel like you are using `MapDynamicControllerRoute` which is for Controller and not for Razor pages. For Razor pages you need to use `MapDynamicPageRoute`

**Reply by scottiemc7** [reply]
Ha, this was a long while back so I had to dig through a bunch of stuff to find what I actually did. I think you're right. I will test that out and see if that fixes the issue. Thanks!

**Reply by scottiemc7** [reply]
Yes, that was it. Thanks again
